src/main/kotlin/de/bund/digitalservice/useid/statics/WidgetController.kt

```diff
@@ -1,15 +1,30 @@
 package de.bund.digitalservice.useid.statics
 
+import de.bund.digitalservice.useid.config.ContentSecurityPolicyProperties
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.http.server.reactive.ServerHttpResponse
 import org.springframework.stereotype.Controller
-import org.springframework.ui.Model
 import org.springframework.web.bind.annotation.GetMapping
 import reactor.core.publisher.Mono
 
 @Controller
-class WidgetController {
+class WidgetController(
+    private val contentSecurityPolicyProperties: ContentSecurityPolicyProperties
+) {
     @GetMapping("/widget")
-    fun widget(model: Model): Mono<String> = Mono.just("widget")
+    fun widget(serverHttpResponse: ServerHttpResponse, serverHttpRequest: ServerHttpRequest): Mono<String> {
+        val host = serverHttpRequest.uri.host
+        val allowedHost = contentSecurityPolicyProperties.domainIsAllowed(host)
+
+        if (allowedHost) {
+            serverHttpResponse.headers.set(
+                "Content-Security-Policy",
+                contentSecurityPolicyProperties.getCSPHeaderValue(host)
+            )
+        }
```

Comment:
Let's keep in mind to put all this to a custom filter in case it is needed for more than one endpoint at some point. For now it is ok to do it here though. 
Do the error pages of the widget need the same CSP header?

Reply:
hmm maybe with webfilter is much easier, but we cannot control per Controller, e.g. we need to define a path in the filter which can also lead to mistake... 
we could however make the path as constant...

Reply:
Yes, we could do that using constants. I think for now it's fine to have it in the Controller code. But let's keep in mind that we should move it to a Filter once we have to reuse it somewhere.